espefuse.py

```diff
@@ -515,26 +524,44 @@ def summary(esp, efuses, args):
                 perms = "-/W"
             else:
                 perms = "-/-"
-            value = str(e.get())
+            base_value = e.get()
+            value = str(base_value)
             if not readable:
                 value = value.replace("0", "?")
-            print(ROW_FORMAT % (e.register_name, e.description, "\n  " if len(value) > 20 else "", value, perms, raw))
-        print("")
-    sdio_force = efuses["XPD_SDIO_FORCE"]
-    sdio_tieh = efuses["XPD_SDIO_TIEH"]
-    sdio_reg = efuses["XPD_SDIO_REG"]
-    if sdio_force.get() == 0:
-        print("Flash voltage (VDD_SDIO) determined by GPIO12 on reset (High for 1.8V, Low/NC for 3.3V).")
-    elif sdio_reg.get() == 0:
-        print("Flash voltage (VDD_SDIO) internal regulator disabled by efuse.")
-    elif sdio_tieh.get() == 0:
-        print("Flash voltage (VDD_SDIO) set to 1.8V by efuse.")
-    else:
-        print("Flash voltage (VDD_SDIO) set to 3.3V by efuse.")
-    warnings = efuses.get_coding_scheme_warnings()
-    if warnings:
-        print("WARNING: Coding scheme has encoding bit error warnings (0x%x)" % warnings)
-
+            if args.format == 'summary':
+                print(ROW_FORMAT % (e.register_name, e.description, "\n  " if len(value) > 20 else "", value, perms, raw),file=export_file)
+            if args.format == 'json':
+                json_efuse[e.register_name]= {
+                    'value': base_value if readable else value,
+                    'readable':readable,
+                    'writeable':writeable }
+        if args.format == 'summary':
+            print("",file=export_file)
+    if args.format == 'summary':
+        sdio_force = efuses["XPD_SDIO_FORCE"]
+        sdio_tieh = efuses["XPD_SDIO_TIEH"]
+        sdio_reg = efuses["XPD_SDIO_REG"]
+        if sdio_force.get() == 0:
+            print("Flash voltage (VDD_SDIO) determined by GPIO12 on reset (High for 1.8V, Low/NC for 3.3V).",file=export_file)
+        elif sdio_reg.get() == 0:
+            print("Flash voltage (VDD_SDIO) internal regulator disabled by efuse.",file=export_file)
+        elif sdio_tieh.get() == 0:
+            print("Flash voltage (VDD_SDIO) set to 1.8V by efuse.",file=export_file)
+        else:
+            print("Flash voltage (VDD_SDIO) set to 3.3V by efuse.",file=export_file)
+        warnings = efuses.get_coding_scheme_warnings()
+        if warnings:
+            print("WARNING: Coding scheme has encoding bit error warnings (0x%x)" % warnings,file=export_file)
+        if args.file:
+            export_file.close()
+            print("Done")
+    if args.format == 'json':
+        if args.file:
+            export_file.write(json.dumps(json_efuse))
+            export_file.close()
+            print("Done")
+        else:
+            print(json.dumps(json_efuse,sort_keys=True,indent=4))
```

Comment:
Suggest doing
```
json.dump(json_efuse, export_file, sort_keys=True, indent=4)
```
rather than doing something different for each case here.

It's less code, and formatting the file in a more friendly way may also be useful.